Modules/fact.py

```diff
@@ -0,0 +1,225 @@
+"""
+fact.py - Fact Class
+
+Provides a Fact object for the Fact Manager Class.
+
+Copyright (c) 2018 The Fuel Rat Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+"""
```

Comment:
May i suggest you look into `dataclass`es, I think this object can be better implemented as one.